third_party/move/move-model/src/ast.rs

```diff
@@ -872,101 +898,106 @@ impl ExpData {
     /// be called before descending into expression, and `visitor(true, ..)` after. Notice
     /// we use one function instead of two so a lambda can be passed which encapsulates mutable
     /// references.
-    pub fn visit_pre_post<F>(&self, visitor: &mut F)
+    /// - `visitor` returns `Err(())`` to indicate that visit should stop early.
+    /// - `visit_pre_post` returns `false` if visitor returned `false`.
+    pub fn visit_pre_post<F>(&self, visitor: &mut F) -> Result<(), ()>
```

Comment:
Why does this function now return a result? We generally do not use Result in this code to enfoster a style of 'keep going' and adding any errors to the global env instead of using the Result mechanism.

Reply:
2nd thought: if this allows us to shortcut if we do a find on the AST, then that's a good use case.

Reply:
Oops, I missed this at first.  Reverting the revert again.